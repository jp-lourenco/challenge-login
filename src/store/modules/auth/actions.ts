export function signInRequest() {
  return {
    type: '@auth/SIGN_IN_REQUEST',
  };
}

export function signInSuccess({ token }: { token: string | null }) {
  return {
    type: '@auth/SIGN_IN_SUCCESS',
    payload: {
      token,
    },
  };
}

export function signInFailure() {
  return {
    type: '@auth/SIGN_IN_FAILURE',
  };
}

export function setEmail({ email }: { email: string }) {
  return {
    type: '@auth/SET_EMAIL',
    payload: {
      email,
    },
  };
}

export function setErrorEmail({ message }: { message: string }) {
  return {
    type: '@auth/SET_ERROR_EMAIL',
    payload: {
      message,
    },
  };
}

export function setPassword({ password }: { password: string }) {
  return {
    type: '@auth/SET_PASSWORD',
    payload: {
      password,
    },
  };
}

export function setErrorPassword({ message }: { message: string }) {
  return {
    type: '@auth/SET_ERROR_PASSWORD',
    payload: {
      message,
    },
  };
}
