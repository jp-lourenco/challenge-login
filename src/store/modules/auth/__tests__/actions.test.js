import * as actions from '../actions';

describe('auth actions', () => {
  test('signInRequest', () => {
    expect(actions.signInRequest()).toMatchSnapshot();
  });

  test('signInSuccess', () => {
    expect(actions.signInSuccess({ token: 'dwijwdji' })).toMatchSnapshot();
  });

  test('signInFailure', () => {
    expect(actions.signInFailure()).toMatchSnapshot();
  });

  test('setEmail', () => {
    expect(actions.setEmail({ email: 'test@example.com' })).toMatchSnapshot();
  });

  test('setErrorEmail', () => {
    expect(
      actions.setErrorEmail({ message: 'Some error message' }),
    ).toMatchSnapshot();
  });

  test('setPassword', () => {
    expect(
      actions.setPassword({ password: 'test@example.com' }),
    ).toMatchSnapshot();
  });

  test('setErrorPassword', () => {
    expect(
      actions.setErrorPassword({ message: 'Some error message' }),
    ).toMatchSnapshot();
  });
});
