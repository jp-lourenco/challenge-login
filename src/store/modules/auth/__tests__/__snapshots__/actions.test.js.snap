// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`auth actions setEmail 1`] = `
Object {
  "payload": Object {
    "email": "test@example.com",
  },
  "type": "@auth/SET_EMAIL",
}
`;

exports[`auth actions setErrorEmail 1`] = `
Object {
  "payload": Object {
    "message": "Some error message",
  },
  "type": "@auth/SET_ERROR_EMAIL",
}
`;

exports[`auth actions setErrorPassword 1`] = `
Object {
  "payload": Object {
    "message": "Some error message",
  },
  "type": "@auth/SET_ERROR_PASSWORD",
}
`;

exports[`auth actions setPassword 1`] = `
Object {
  "payload": Object {
    "password": "test@example.com",
  },
  "type": "@auth/SET_PASSWORD",
}
`;

exports[`auth actions signInFailure 1`] = `
Object {
  "type": "@auth/SIGN_IN_FAILURE",
}
`;

exports[`auth actions signInRequest 1`] = `
Object {
  "type": "@auth/SIGN_IN_REQUEST",
}
`;

exports[`auth actions signInSuccess 1`] = `
Object {
  "payload": Object {
    "token": "dwijwdji",
  },
  "type": "@auth/SIGN_IN_SUCCESS",
}
`;
